src/pretix/control/views/waitinglist.py

```diff
@@ -366,3 +427,87 @@ def get_success_url(self) -> str:
             'event': self.request.event.slug,
             'organizer': self.request.event.organizer.slug
         })
+
+class EntryUpdate(EventPermissionRequiredMixin, UpdateView):
+    model = WaitingListEntry
+    template_name = 'pretixcontrol/waitinglist/update.html'
+    permission = 'can_change_orders'
+    form_class = WaitingListEntryEditForm
+    context_object_name = 'entry'
+
+
+
+    #def dispatch(self, request, *args, **kwargs):
+        #self.fields['subevent'] = WaitingListEntry.objects.filter(event=self.request.event)
+    #    return super().dispatch(request, *args, **kwargs)
+    def get_queryset(self):
+        return WaitingListEntry.objects.filter(
+            event=self.request.event,
+        )
+    def get_object(self, queryset=None) -> WaitingListEntry:
+        try:
+            return self.request.event.waitinglistentries.get(
+                id=self.kwargs['entry'],
+                event=self.request.event,
+                voucher__isnull=True,
+            )
+        except WaitingListEntry.DoesNotExist:
+            raise Http404(_("The requested entry does not exist."))
+
+    def get_success_url(self) -> str:
+        return reverse('control:event.orders.waitinglist', kwargs={
+            'event': self.request.event.slug,
+            'organizer': self.request.event.organizer.slug
+        })
+
+    @transaction.atomic
+    def update(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        success_url = self.get_success_url()
+        self.object.log_action('pretix.event.orders.waitinglist.transfered', user=self.request.user)
+        messages.success(self.request, _('The selected entry has been transfered.'))
+        if "next" in self.request.GET and is_safe_url(self.request.GET.get("next"), allowed_hosts=None):
+            return redirect(self.request.GET.get("next"))
+        return HttpResponseRedirect(success_url)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['event'] = self.request.event
+        return kwargs
```

Comment:
This is already done by `EventBasedFormMixin`, which `UpdateView` inherits from.

```suggestion
```

Reply:
I think this exists for a reason - leaving it out results in 'NoneType' object has no attribute 'subevents' in form class WaitingListEntryEditForm for "self.event.subevents.all()" . The standart method doesnt seem to hand over self.event